### PR TITLE
Support multi-versions and multi-targets libtorch

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: torchx/README.md [hunk 1]
```diff
@@ -27,13 +27,22 @@ end
...
 
 We will automatically download a precompiled version of `LibTorch` that runs on the CPU.
-If you want to use another version, [download the .zip file of another `LibTorch` version
-from the official website](https://pytorch.org/get-started/locally/), unpack it to a
-directory of your choice and set the `LIBTORCH_DIR` [environment variable](https://en.wikipedia.org/wiki/Environment_variable)
-to point to the unpacked directory.
-
-Once downloaded, we will compile `Torchx` bindings. You will need `make`, `cmake` (3.12+)
-and a `C++` compiler. If running on Windows, you will need:
+If you want to use another version, please use LIBTORCH_VER to choose the different versions. The current
```
Comment (josevalim) [inline torchx/README.md:30]:
```suggestion
If you want to use another version, please use `LIBTORCH_VERSION` to choose the different versions. The current
```

Can we please use `LIBTORCH_VERSION` as the variable name? No abbreviations:

File: torchx/README.md [hunk 2]
```diff
@@ -27,13 +27,22 @@ end
...
+supported versions are:
+- 1.9.0
+- 1.9.1
+- 1.10.0
+- 1.10.1
+- 1.10.2
+
+If you want to use the torch with CUDA support, please use LIBTORCH_TARGET to choose CUDA version. The current
```
Comment (josevalim) [inline torchx/README.md:38]:
```suggestion
If you want to use the torch with CUDA support, please use `LIBTORCH_TARGET` to choose CUDA version. The current
```

File: torchx/mix.exs [hunk 3]
```diff
@@ -97,6 +92,11 @@ defmodule Torchx.MixProject do
         raise "No CUDA support on OSX"
```
Comment (josevalim) [inline torchx/mix.exs:92]:
```suggestion
        Mix.error("No CUDA support on OSX")
```

File: torchx/mix.exs [hunk 4]
```diff
@@ -97,6 +92,11 @@ defmodule Torchx.MixProject do
...
       end
 
+      # Check if target is valid
+      unless Enum.member?(@valid_targets, @libtorch_target) do
+        raise "Invalid target, please use one of #{@valid_targets}"
```
Comment (josevalim) [inline torchx/mix.exs:97]:
```suggestion
        Mix.error("Invalid target, please use one of #{inspect(@valid_targets)}")
```

File: torchx/mix.exs [hunk 5]
```diff
@@ -4,19 +4,14 @@ defmodule Torchx.MixProject do
...
-    @libtorch_dir libtorch_dir
-    @libtorch_compilers [:elixir_make]
-  else
-    @libtorch_base "libtorch-#{@libtorch_version}-#{@libtorch_target}"
-    @libtorch_dir Path.join(__DIR__, "cache/#{@libtorch_base}")
-    @libtorch_compilers [:torchx, :elixir_make]
-  end
+  @valid_targets ["cpu", "cu102", "cu111"]
+
+  @libtorch_version System.get_env("LIBTORCH_VER", "1.9.1")
```
Comment (josevalim) [inline torchx/mix.exs:9]:
```suggestion
  @libtorch_version System.get_env("LIBTORCH_VERSION", "1.9.1")
```

File: torchx/README.md (hunk 1 above)
Comment (howard0su) [reply torchx/README.md:30]: sure

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**